### ISSUE TITLE
Page uses a file .txt (that end with the same word from a real blueprint) instead of the right one

### ISSUE THREAD

**Issue by jesuismaxime** [getkirby/kirby#2659]
**Describe the bug**  
I have a page where I store all my site images that use this blueprint(bp): `ressources-images.txt`, a simple file listing, but the page is using the default blueprint instead. 

So, using the panel, I put some random content in this page (since my default template have a `textarea` field), save it and found that random string in a file named `stasher_page_artisan.jpg.fr.txt`. I found that the file is actually missing, so the `.jpg.{lang}.txt` is used as the page blueprint. 

The "matching blueprint/page template" function should skip filenames that contain a file extension to avoid that kind of bugs.

**Comment by lukasbestle** [comment]
I get why this can be an issue. Theoretically it should never occur as the Panel deletes the text files when the files are deleted, but of course that can fail or it can also happen when files are deleted manually.

The question is: How should Kirby define a "filename that contains a file extension"? Using dots in blueprint names is allowed, so there are users who use blueprints like `article.image`. A file called `article.text.txt` should therefore *not* be skipped.

Maybe we could check if the "extension" is a common one like `jpg`, `png` or `pdf` to cover at least the most common cases? But that would prevent people from having common file extensions in their blueprint names (which until then was completely valid).

Another option: For text files that contain a dot in the blueprint name, Kirby could check if that blueprint actually exists and only use the file if it does. That would however also be a breaking change for those who don't use blueprints at all.

So to summarize: Pretty difficult to implement. 😞

**Comment by neildaniels** [comment]
You could change the file extension that Kirby uses for it's own content files. https://getkirby.com/docs/reference/system/options/content#content-file-extension

Alternatively, you might be able to add your `ressources-images.txt` to the ignore list. https://getkirby.com/docs/reference/system/options/content#ignore-files-folders

**Comment by jesuismaxime** [comment]
@lukasbestle :
A page template could be easily matched/find with the `{slugged-page}.{kirby ext content file}` (or `{slugged-page}.{locale}.{kirby ext content file}` for multi-language site).

A page slug cannot contain a dot `.` so it's basically impossible to have a page with this id `stasher_page_artisan.jpg`

@neildaniels since the content file extension is changeable, the image content file will also be using that extension, so your point is off track..

Why would I want to ignore my content file ? The whole purpose of this file is to set my template. The whole issue is about that. If an orphan `.txt`* (from an error while deleting a file) is left alone Kirby may use that file as the template instead of using the right one.  

* or whatever your base content file extension is

**Comment by lukasbestle** [comment]
The content filename is not the slug, but the blueprint/template name. As I explained in my comment above, the blueprint/template name *can* contain dots (and users do actually use this for their sites).

**Comment by jesuismaxime** [comment]
@lukasbestle yeah sorry, mixed up while I was talking about the filename. 

Bottom line : There's a need for protected values like for the field handle (Got issues whileback between K2 and K3 with fields named 'content'), most common image, video and doc files should be in it. If it's not possible, maybe a warning system (red notice in the panel page, etc) to debug easily (the Ressources directory I was talking about earlier contains 600 images (so 1200 files) and I found 4 .txt orphans

**Comment by lukasbestle** [comment]
We are talking about two different issues now though, aren't we? The original issue was that Kirby uses the content file of a file as the content file for the page. What you are referring to now is that field names can't be the reserved names of other methods. We had a discussion about that somewhere, but I can't find it right now.

But please let's only discuss one issue per issue.

**Comment by distantnative** [comment]
While I understand the initial issue, as @lukasbestle described, we cannot base it on any pattern based on separating dots - blueprint/template names often feature dots. 

An idea, not sure if that could work:
When scanning the directory, can we go through all actual files found, filter all content files that match `original Filename.originalFileExtension.lang.txt` - and only pick the page content files from the remaining `txt` files?

**Comment by distantnative** [comment]
In fact, I think that is what https://github.com/getkirby/kirby/blob/master/src/Cms/Dir.php#L139-L142 is supposed to do.

**Comment by lukasbestle** [comment]
@distantnative Yes, that's what we do at the moment. It only fails when the file is deleted, but the file meta file is still present (like in the example in https://github.com/getkirby/kirby/issues/2659#issue-646693898).

To be honest, I think this edge-case is unfixable for the reasons mentioned.
